src/BreakpointViewer.cpp

```diff
@@ -856,22 +851,24 @@ std::optional<Breakpoint> BreakpointViewer::parseTableRow(BreakpointRef::Type ty
 	return bp;
 }
 
-
 void BreakpointViewer::onSymbolTableChanged()
 {
 	for (int row = 0; row < bpTableWidget->rowCount(); ++row) {
-		auto* item = bpTableWidget->item(row, BP_ADDRESS);
+		auto* item = bpTableWidget->item(row, LOCATION);
 
-		// scan tooltip validity
-		if (!item->text().isEmpty()) {
-			Symbol* s = debugSession.symbolTable().getAddressSymbol(item->text());
-			auto address = stringToValue<uint16_t>(item->text());
-			assert(address);
+		if (auto* ref = scanBreakpointRef(BreakpointRef::BREAKPOINT, row)) {
+			const auto& bp = breakpoints->getBreakpoint(ref->index);
+			QStringList labels = debugSession.symbolTable().findAddressSymbolsByAddress(bp.range->start);
+			auto addressStr = hexValue(bp.range->start, 4);
 
-			if (!s || *address != s->value()) {
-				setTextField(BreakpointRef::BREAKPOINT, row, LOCATION, item->text());
-				setTextField(BreakpointRef::BREAKPOINT, row, BP_ADDRESS, "");
-			}
+			setTextField(BreakpointRef::BREAKPOINT, row, LOCATION,
+			             labels.isEmpty() ? addressStr :
+			             labels.contains(item->text()) ? item->text() : labels.at(0),
+			             labels.isEmpty() ? "" : addressStr);
+		}
+		else {
```

Comment:
How can it be that we don't find a `BreakpointRef`? This should only be possible for incomplete rows I think? (But I must admit the whole mapping from tables, via `maps` to `Breakpoint` objects in `Breakpoints` in not 100% clear to me).

Shouldn't such incomplete rows be left untouched by this method? This might also be related to the discussion above about typos (and thus incomplete rows in general).